commit 49d52b547fe1e66aac5b0799c6918ae3a5a831da
Author: Miroslav Lichvar <mlichvar@redhat.com>
Date:   Fri Aug 31 12:48:07 2018 +0200

    test: make 129-reload more reliable

diff --git a/test/simulation/129-reload b/test/simulation/129-reload
index 9c4e985..7a346a4 100755
--- a/test/simulation/129-reload
+++ b/test/simulation/129-reload
@@ -9,7 +9,8 @@ limit=100
 min_sync_time=100
 max_sync_time=104
 client_chronyd_options="-r"
-client_conf="dumpdir tmp"
+client_conf="dumpdir tmp
+maxupdateskew 10000"
 
 run_test || test_fail
 
