diff -up chrony-2.4/examples/chrony-wait.service.chrony-wait chrony-2.4/examples/chrony-wait.service
--- chrony-2.4/examples/chrony-wait.service.chrony-wait	2016-06-07 11:20:59.000000000 +0200
+++ chrony-2.4/examples/chrony-wait.service	2016-10-27 14:05:14.030882058 +0200
@@ -9,7 +9,7 @@ Wants=time-sync.target
 Type=oneshot
 # Wait up to ~10 minutes for chronyd to synchronize and the remaining
 # clock correction to be less than 0.1 seconds
-ExecStart=/usr/bin/chronyc waitsync 600 0.1 0.0 1
+ExecStart=/usr/bin/chronyc -h 127.0.0.1,::1 waitsync 600 0.1 0.0 1
 RemainAfterExit=yes
 StandardOutput=null
 
