From f1a74e41b3b89771243294097ebd3472deb019f4 Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Thu, 6 Nov 2008 17:35:22 +0100
Subject: [PATCH] Fix resolving IP addresses into names on 64-bit big endian machines

---
 nameserv.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/nameserv.c b/nameserv.c
index f509dfb..dd610df 100644
--- a/nameserv.c
+++ b/nameserv.c
@@ -65,7 +65,7 @@ DNS_IPAddress2Name(unsigned long ip_addr)
   struct hostent *host;
   static char buffer[16];
   unsigned int a, b, c, d;
-  unsigned long addr;
+  uint32_t addr;
 
   addr = htonl(ip_addr);
   if (addr == 0UL) {
-- 
1.5.6.5

