--- chrony-1.23/io_linux.h.orig	2009-06-08 09:25:23.000000000 +0200
+++ chrony-1.23/io_linux.h	2009-06-08 09:26:24.000000000 +0200
@@ -6,7 +6,8 @@
 
 /* Hmm.  These constants vary a bit between systems. */
 /* (__sh__ includes both sh and sh64) */
-#if defined(__i386__) || defined(__sh__) || defined(__arm__)||defined(__x86_64__)
+/* (__s390__ includes both s390 and s390x) */
+#if defined(__i386__) || defined(__sh__) || defined(__arm__)||defined(__x86_64__) || defined(__s390__)
 #define CHRONY_IOC_NRBITS	8
 #define CHRONY_IOC_TYPEBITS	8
 #define CHRONY_IOC_SIZEBITS	14
