From bcedacaa3d9e51d5c1dfb1d4614fb4ced6516382 Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Mon, 10 Sep 2018 14:51:30 +0200
Subject: test: fix samplefilt unit test to work with low-precision clock

Origin: upstream, https://git.tuxfamily.org/chrony/chrony.git/commit/?id=bcedacaa3d9e51d5c1dfb1d4614fb4ced6516382
Bug: https://listengine.tuxfamily.org/chrony.tuxfamily.org/chrony-dev/2018/09/msg00000.html
Last-Update: 2018-09-10
---
 test/unit/samplefilt.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/test/unit/samplefilt.c
+++ b/test/unit/samplefilt.c
@@ -18,9 +18,13 @@
  **********************************************************************
  */
 
-#include <samplefilt.c>
+#include <local.h>
 #include "test.h"
 
+#define LCL_GetSysPrecisionAsQuantum() (1.0e-6)
+
+#include <samplefilt.c>
+
 void
 test_unit(void)
 {
